Move create to Axis.prototype

pcasaretto opened the pull request:
This allows recursive creation of axis instances.
This makes the API very concise, and code that uses axios clients can specialize each client further without knowledge if it is a "root" client or not.

Wanted to make sure this idea is interesting before commiting time to add tests.
Also, this is my first JS PR ever.

yasuf commented:
I appreciate the interest and the idea. We're fine with the implementation as is and this might need plenty of testing about how each part of the configuration works in the recursively created instance. If we can avoid complexity, we should. Do you have a use case for this?

pcasaretto commented:
I do have one.
We have an internal API with it's own authentication scheme and I've refactored a lot of internal code recently to make sure all the JS code that is interacting with this API is using the same implementation.
Some parts of the code used `axios.create` followed by a change in `defaults` to set some headers appropriate to that specific feature. When I refactored the code to use the proper client (an axios instance) this code broke and I was forced to manually include the headers in all calls.

pcasaretto commented:
Favoring #2040
